Tests/MongoSwiftSyncTests/SyncMongoClientTests.swift

```diff
@@ -320,4 +320,35 @@ final class SyncMongoClientTests: MongoSwiftTestCase {
         expect(try MongoClient("mongodb://localhost:12345", options: options))
             .to(throwError(errorType: MongoError.InvalidArgumentError.self))
     }
+
+    func testConnectionTimeout() throws {
+        var failOptions = MongoClientOptions()
+        failOptions.connectTimeoutMS = 100
+        failOptions.serverSelectionTimeoutMS = 1000
+
+        var succeedOptions = MongoClientOptions()
+        succeedOptions.connectTimeoutMS = 1000
+        succeedOptions.serverSelectionTimeoutMS = 1000
+
+        let setupClient = try MongoClient.makeTestClient()
+        let timeoutClient = try MongoClient.makeTestClient(options: failOptions)
+        let succeedClient = try MongoClient.makeTestClient(options: succeedOptions)
+
+        guard try setupClient.supportsFailCommand() else {
```

Comment:
nit: I would put the setupClient initialization and check for failCommand support first in this test before the rest of the code, so we can "fail fast"/exit quickly and the exit conditions are easier to spot at a quick glance.